script.js

```diff
@@ -0,0 +1,256 @@
+       element = element[0];
+       element.classList.remove('keypress');
+       if (element.innerText == 'Backspace') {
+         output = output.substr(0, output.length-1);
+       } else if (element.innerText == 'Enter') {
+         output += '\n';
+       } else if (element.innerText == '') {
+         output += ' ';
+       } else if (element.innerText == 'Tab') {
+         output += '   ';
+       } else if (element.innerText == 'Caps Lock') {
+         output += '';
+       } else {
+         output += element.innerText;
+       }
+       document.querySelector('textarea').value = output;
+     } else {
+       element = event.target;
+       element.classList.remove('keypress');
+     }
+     let keyCode = event.target.getAttribute('data');
+     if (keyCode == 20) {
+       caps = !caps;
+       let langKeys;
+       isEng == 1 ? langKeys = engKeysUpperCase : langKeys = rusKeysUpperCase;
+       let i = 0;
+       let keys = document.querySelectorAll('#keyboard div');
+       if (caps) {
+         element.classList.add('keypress');
+         keys.forEach((el) => {
+           el.innerText = langKeys[i];
+           i++;
+         })
+       } else {
+         isEng == 1 ? langKeys = engKeys : langKeys = rusKeys;
+         element.classList.remove('keypress');
+         keys.forEach((el) => {
+           el.innerText = langKeys[i];
+           i++;
+         });
+       }
+     }
+   });
+  });
+
+  document.addEventListener('keyup', function(event) {
+    if (event.code == 'ShiftLeft') leftShift = false;
+    else if (event.code == 'AltLeft') {
+      if (leftShift) {
+        isEng == 1 ? isEng = 0 : isEng = 1;
+        let l = document.querySelector('.language');
+        isEng == 1 ? l.innerText = 'Eng' : l.innerText = 'Rus';
+        let i = 0;
+        let keys = document.querySelectorAll('#keyboard div');
+        let langKeys;
+        isEng == 1 ? langKeys = engKeys : langKeys = rusKeys;
+        keys.forEach((el) => {
+          el.innerText = langKeys[i];
+          i++;
+        });
+      }
+    }
+    let element = document.querySelectorAll('div[data="' + `${event.keyCode}` + '"]');
+    if (element.length == 1) {
+      element = element[0];
+      element.classList.remove('keypress');
+    } else {
+      if (event.code == "ShiftLeft") element[0].classList.remove('keypress');
+      else if (event.code == "ShiftRight") element[1].classList.remove('keypress');
+      else if (event.code == "ControlLeft") element[0].classList.remove('keypress');
+      else if (event.code == "ControlRight") element[1].classList.remove('keypress');
+      else if (event.code == "MetaLeft") element[0].classList.remove('keypress');
+      else if (event.code == "MetaRight") element[1].classList.remove('keypress');
+      else if (event.code == "AltLeft") element[0].classList.remove('keypress');
+      else if (event.code == "AltRight") element[1].classList.remove('keypress');
+    }
+    if (event.keyCode == 20) {
```

Comment:
Please comment, why 20?
example
// The 20 is the keyCode of ...

Reply:
edited